### PR TITLE
docs: add rule documentation (#581)

### LINKED ISSUE #522: Validate trip frequencies do not overlap
**Describe the solution you'd like**
Validate trip frequencies do not overlap

**Additional context**
`v1.4.0` implementation:
- https://github.com/MobilityData/gtfs-validator/blob/v1.4.0/usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateTripFrequenciesOverlap.java
- https://github.com/MobilityData/gtfs-validator/blob/v1.4.0/usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateTripFrequenciesOverlapTest.java

### PR DESCRIPTION
**Pull request by lionel-nj**
closes #522 

**Summary:**

This PR adds information to `RULES.md`: documentation about the ryule implemented in #581.

**Expected behavior:** 

No code change.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: -new feature short description-" (PR title must follow the [conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0/))
- [x] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: RULES.md [hunk 1]
```diff
@@ -256,6 +256,29 @@ Any other combination raise this error.
...
+
+### OverlappingTripFrequenciesNotice
+
+Trip frequencies must not overlap in time
+
+#### References:
+
+* [GTFS frequencies.txt specification](http://gtfs.org/reference/static/#frequenciestxt)
+
+Value of field `location_type` of parent found in field `parent_station` is invalid.
```
Comment (barbeau) [inline RULES.md:269]: I think this is missing an error header, or this text shouldn't be here?

It's about parent_stations, but under the header for OverlappingTripFrequenciesNotice

**Review by barbeau — CHANGES_REQUESTED** [review]
@lionel-nj Thanks! See below.

File: RULES.md (hunk 1 above)
Comment (lionel-nj) [reply RULES.md:269]: Indeed, thanks! Modifying this.

**Review by barbeau — APPROVED** [review]
👍

**Pull request merged**